design-library/src/components/BccTooltip/BccTooltip.vue

```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+type Props = {
+  primaryPosition?: "top" | "bottom";
+  secondaryPosition?: "left" | "center" | "right";
+  variant?: "dark" | "white" | "grey";
+  text: string;
+  visible?: boolean;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  primaryPosition: "top",
+  secondaryPosition: "center",
+  variant: "dark",
+  text: "",
+  visible: false,
+});
+</script>
+
+<template>
+  <div class="tooltip-wrapper">
+    <slot></slot>
+    <div
+      :class="[
+        'tooltip-content',
+        `${props.primaryPosition}-${props.secondaryPosition}`,
+        props.variant,
+        { visible: props.visible },
+      ]"
+    >
+      {{ props.text }}
```

Comment:
Suggestion: Use a slot instead so any arbitrary content can be passed to it. Most other components also use slots over props for text content (but it's not a hard rule or anything). I guess it might come down a bit to personal preference, I like slots 😅

Reply:
Done ✅  - I used two slots - one is to wrap the component I want to display the tooltip over, the other one is just a text, the one to wrap is the named slot so users can either wrap it or use how they want